### ISSUE TITLE
Functions Reference - Array reductions contain vector-only signatures

### ISSUE THREAD

**Issue by andrjohns** [stan-dev/docs#758]
Not sure if this is intended or worth changing, but the following manual sections for Array operations include signatures with no array types:

- [Norms](https://mc-stan.org/docs/functions-reference/array_operations.html#norms): Contains both `array` and `vector`/`row_vector` signatures
- [Euclidean/Squared distance](https://mc-stan.org/docs/functions-reference/array_operations.html#euclidean-distance-and-squared-distance): Contains no `array` signatures at all

#### Current Version:
v2.18.0

**Comment by WardBrian** [comment]
Distance functions seem to definitely be on the wrong page. The norms could be split, like we have for a lot of functions, but we've also talked about re-thinking doing that because it might mean the first page you see only lists half the signatures and you think that's all there are.

**Comment by andrjohns** [comment]
> The norms could be split, like we have for a lot of functions, but we've also talked about re-thinking doing that because it might mean the first page you see only lists half the signatures and you think that's all there are.

Yeah completely agree, that's why I wasn't completely sure if it should be considered a bug or not
